sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py

```diff
@@ -138,477 +145,771 @@ def _get_data_file(self) -> str:
         """Return the test file name."""
         return os.path.join(os.path.dirname(os.path.dirname(__file__)), "test_data", "product_info_1.md")
 
-    # for debugging purposes: if a test fails and its agent has not been deleted, it will continue to show up in the agents list
-    """
-    # NOTE: this test should not be run against a shared resource, as it will delete all agents
+
+
+    # **********************************************************************************
+    #
+    #                      HAPPY PATH SERVICE TESTS - agent APIs
+    #
+    # **********************************************************************************
+
     @agentClientPreparer()
     @recorded_by_proxy
-    def test_clear_client(self, **kwargs):
+    def test_create_client(self, **kwargs):
+        """test client creation"""
+
         # create client
         client = self.create_client(**kwargs)
         assert isinstance(client, AIProjectClient)
 
-        # clear agent list
-        agents = client.agents.list_agents().data
-        for agent in agents:
-            client.agents.delete_agent(agent.id)
-        assert client.agents.list_agents().data.__len__() == 0
-       
         # close client
         client.close()
-    """
 
-    #     # **********************************************************************************
-    #     #
-    #     #                               UNIT TESTS
-    #     #
-    #     # **********************************************************************************
+    
+    @agentClientPreparer()
+    @recorded_by_proxy
+    def test_create_delete_agent(self, **kwargs):
+        """test agent creation and deletion"""
+        # create client
+        # client = self.create_client(**kwargs)
+        with self.create_client(**kwargs) as client:
+            assert isinstance(client, AIProjectClient)
+            print("Created client")
+            self._do_test_create_agent(client=client, body=None, functions=None)
+        
 
-    # # **********************************************************************************
-    # #
-    # #                      HAPPY PATH SERVICE TESTS - agent APIs
-    # #
-    # # **********************************************************************************
+    @agentClientPreparer()
+    @recorded_by_proxy
+    def test_create_agent_with_body(self, **kwargs):
+        """test agent creation with body: JSON"""
+        # create client
+        with self.create_client(**kwargs) as client:
+            assert isinstance(client, AIProjectClient)
+            print("Created client")
+
+            # create body for agent and call helper function
+            body = {
+                "name": "my-agent",
+                "model": "gpt-4o",
```

Comment:
I load model deployment name from environment variable, so it can easily be updated or tested locally against different models. Similar to other test configurations. Can this be done here?